qualtran/_infra/controlled_test.py

```diff
@@ -385,62 +366,6 @@ def test_controlled_global_phase_tensor():
     np.testing.assert_allclose(bloq.tensor_contract(), should_be)
 
 
-@attrs.frozen
-class TestCtrlStatePrepAnd(Bloq):
-    """Decomposes into a Controlled-AND gate + int effects & targets where ctrl is active.
-
-    Tensor contraction should give the output state vector corresponding to applying an
-    `And(and_ctrl)`; assuming all the control bits are active.
-    """
-
-    ctrl_spec: CtrlSpec
-    and_ctrl: Tuple[int, int]
```

Comment:
Can we add a private flag to the `Controlled` class, something of the form `_THRU_REGISTERS_ONLY = True`, and keep this test around? Since we plan to relax this constraint later, it'll be nice to not delete the test.